src/sentry/tasks/weekly_escalating_forecast.py

```diff
@@ -55,15 +66,24 @@ def run_escalating_forecast() -> None:
     default_retry_delay=60,
 )
 @retry
-def generate_forecasts_for_projects(project_ids: List[int]) -> None:
+def generate_forecasts_for_projects(project_ids: List[int], forecast_for_ongoing=False) -> None:
+    query_for_archived_until_escalating = Q(
+        status=GroupStatus.IGNORED, substatus=GroupSubStatus.UNTIL_ESCALATING
+    )
+    query_for_ongoing = Q(status=GroupStatus.UNRESOLVED, substatus=GroupSubStatus.ONGOING)
+
+    combined_query = (
+        (query_for_archived_until_escalating | query_for_ongoing)
+        if forecast_for_ongoing
+        else query_for_archived_until_escalating
+    )
+
     for until_escalating_groups in chunked(
         RangeQuerySetWrapper(
             Group.objects.filter(
-                status=GroupStatus.IGNORED,
-                substatus=GroupSubStatus.UNTIL_ESCALATING,
                 project_id__in=project_ids,
                 last_seen__gte=datetime.now() - timedelta(days=7),
```

Comment:
Do we have `last_seen` indexed? This may make this query very slow; no?

If it's not indexed, we may be better off adding an index OR iterating over the groups and excluding old groups. I'm trying to use CPU on the worker rather than on the DB.